wait/host_port_test.go

```diff
@@ -508,3 +508,58 @@ func TestHostPortStrategySucceedsGivenShellIsNotInstalled(t *testing.T) {
 	err = wg.WaitUntilReady(context.Background(), target)
 	require.NoError(t, err)
 }
+
+func TestHostPortStrategySucceedsGivenShellIsNotFound(t *testing.T) {
+	listener, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer listener.Close()
+
+	rawPort := listener.Addr().(*net.TCPAddr).Port
+	port, err := nat.NewPort("tcp", strconv.Itoa(rawPort))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	target := &MockStrategyTarget{
```

Comment:
suggestion: Could we use the new generated mock `mockStrategyTarget` for this, as we're in the process of moving across to it. You can see an [example in this PR](https://github.com/testcontainers/testcontainers-go/pull/2773).

Reply:
@stevenh it would be nice if you could modify the newly added tests in your PR itself as you have context of the changes. This will help in keeping similar changes in the same PR. wdyt?

Reply:
I can but I'm trying to avoid a moving target and the functionality already exists, its just not in common use yet.

Reply:
some `{}_test.go` files in `/wait` have package = `wait_test` while others have only `wait`. What convention do we want to follow?
This causes unnecessary confusion as `MockStrategyTarget` is in package `wait` while `mockStrategyTarget` is in `wait_test`.

Reply:
I don't understand how `MockStrategyTarget` will be migrated to `mockStrategyTarget`. Could you share a sample for it?
MockStrategyTarget was quite easy to understand to be honest. I'm not too sure which example you're talking about in this PR: https://github.com/testcontainers/testcontainers-go/pull/2773

Reply:
You can see an example in the wait file tests here https://github.com/testcontainers/testcontainers-go/blob/main/wait/file_test.go

Reply:
Done.